MapboxCoreNavigationTests/FreeDriveLocationManager.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import MapboxNavigationNative
+import TestHelper
+import XCTest
+@testable import MapboxCoreNavigation
+import MapboxAccounts
+
+class Road {
+    let from: CLLocationCoordinate2D
+    let to: CLLocationCoordinate2D
+    let length: Double
+
+    init(from: CLLocationCoordinate2D, to: CLLocationCoordinate2D) {
+        self.from = from
+        self.to = to
+        self.length = (to - from).length()
+    }
+
+    func whichCloser(a: CLLocationCoordinate2D, b: CLLocationCoordinate2D) -> CLLocationCoordinate2D {
+        return proximity(of: a) > proximity(of: b) ? b : a
+    }
+
+    func proximity(of: CLLocationCoordinate2D) -> Double {
+        return ((of - from).length() + (of - to).length()) / length
+    }
+}
+
+extension CLLocationCoordinate2D {
+    static func -(a: CLLocationCoordinate2D, b: CLLocationCoordinate2D) -> CLLocationCoordinate2D {
+        return CLLocationCoordinate2D(latitude: a.latitude - b.latitude, longitude: a.longitude - b.longitude)
+    }
+
+    func length() -> Double {
+        return sqrt(latitude * latitude + longitude * longitude)
+    }
+}
+
+class FreeDriveLocationManagerTests: XCTestCase {
+    class Delegate: FreeDriveLocationManagerDelegate {
+        let road: Road
+        let locationUpdateExpectation: XCTestExpectation
+        
+        init(road: Road, locationUpdateExpectation: XCTestExpectation) {
+            self.road = road
+            self.locationUpdateExpectation = locationUpdateExpectation
+        }
+        
+        func locationManager(_ manager: FreeDriveLocationManager, didUpdateLocation location: CLLocation, rawLocation: CLLocation) {
+            print("Got location: \(rawLocation.coordinate.latitude), \(rawLocation.coordinate.longitude) → \(location.coordinate.latitude), \(location.coordinate.longitude)")
+            print("Value: \(road.proximity(of: location.coordinate)) < \(road.proximity(of: rawLocation.coordinate))")
```

Comment:
Once we determine an acceptable range of values, we should turn these print statements into assertions.